java/ql/lib/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -182,6 +204,695 @@ private class ApacheKeyValueModel extends SummaryModelCsv {
+          ".bidimap;AbstractDualBidiMap;true;AbstractDualBidiMap;;;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;AbstractDualBidiMap;;;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;AbstractDualBidiMap;;;MapKey of Argument[1];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;AbstractDualBidiMap;;;MapValue of Argument[1];MapKey of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;AbstractDualBidiMap;;;MapKey of Argument[2];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;AbstractDualBidiMap;;;MapValue of Argument[2];MapKey of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;createBidiMap;;;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;createBidiMap;;;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;createBidiMap;;;MapKey of Argument[1];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;createBidiMap;;;MapValue of Argument[1];MapKey of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;createBidiMap;;;MapKey of Argument[2];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;createBidiMap;;;MapValue of Argument[2];MapKey of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap;true;createKeySetIterator;;;MapKey of Argument[-1];Element of ReturnValue;value",
+          ".bidimap;AbstractDualBidiMap;true;createValuesIterator;;;MapValue of Argument[-1];Element of ReturnValue;value",
+          ".bidimap;AbstractDualBidiMap;true;createEntrySetIterator;;;MapKey of Argument[-1];MapKey of Element of ReturnValue;value",
+          ".bidimap;AbstractDualBidiMap;true;createEntrySetIterator;;;MapValue of Argument[-1];MapValue of Element of ReturnValue;value",
+          ".bidimap;AbstractOrderedBidiMapDecorator;true;AbstractOrderedBidiMapDecorator;;;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".bidimap;AbstractOrderedBidiMapDecorator;true;AbstractOrderedBidiMapDecorator;;;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".bidimap;AbstractSortedBidiMapDecorator;true;AbstractSortedBidiMapDecorator;;;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".bidimap;AbstractSortedBidiMapDecorator;true;AbstractSortedBidiMapDecorator;;;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".bidimap;DualHashBidiMap;true;DualHashBidiMap;;;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".bidimap;DualHashBidiMap;true;DualHashBidiMap;;;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".bidimap;DualHashBidiMap;true;DualHashBidiMap;;;MapKey of Argument[1];MapValue of Argument[-1];value",
```

Comment:
These have `createBidiMap` methods too, though they're protected so don't matter much (but we are covering protected constructors, so might want to pick them up for consistency)